mobile/examples/whisper/android/azure/app/src/main/java/ai/onnxruntime/example/speechrecognition/SpeechRecognizer.kt

```diff
@@ -0,0 +1,85 @@
+package ai.onnxruntime.example.speechrecognition
+
+import ai.onnxruntime.OnnxTensor
+import ai.onnxruntime.OrtEnvironment
+import ai.onnxruntime.OrtSession
+import ai.onnxruntime.extensions.OrtxPackage
+import android.os.SystemClock
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import java.lang.Exception
+
+@Serializable
+data class Error(
+    val message: String,
+    val type: String,
+    val code: String
+)
+
+@Serializable
+data class Response(
+    val text: String,
+)
+
+@Serializable
+data class ErrorResponse(
+    val error: Error
+)
+
+class SpeechRecognizer(modelBytes: ByteArray) : AutoCloseable {
+    private val session: OrtSession
+    private val baseInputs: Map<String, OnnxTensor>
+
+    init {
+        val env = OrtEnvironment.getEnvironment()
+        val sessionOptions = OrtSession.SessionOptions()
+        sessionOptions.registerCustomOpLibrary(OrtxPackage.getLibraryPath())
+
+        session = env.createSession(modelBytes, sessionOptions)
+
+        // NOTE!! Modify the following line and input your own OpenAI AUTH_TOKEN for successful API request calls.
+        // By default, the behavior of the app will fail to build if no correct authToken provided.
+        val authToken = "Set this to your auth token (OpenAI API Key);"
+        val authTokenInput = OnnxTensor.createTensor(env, arrayOf(authToken), tensorShape(1.toLong()))
+        baseInputs = mapOf(
+            "auth_token" to authTokenInput
+        )
+    }
+
+    data class Result(val text: String, val inferenceTimeInMs: Long)
+
+    fun run(audioTensor: OnnxTensor): Result {
+        val inputs = mutableMapOf<String, OnnxTensor>()
+        baseInputs.toMap(inputs)
+        inputs["transcribe0/file"] = audioTensor
+        val startTimeInMs = SystemClock.elapsedRealtime()
+        val outputs = session.run(inputs)
+        val elapsedTimeInMs = SystemClock.elapsedRealtime() - startTimeInMs
+        val recognizedText = outputs.use {
+            @Suppress("UNCHECKED_CAST")
+            (outputs[0].value as Array<String>)[0]
+        }
+
+        // Parse the response from OpenAI Whisper Endpoint
+        val json = Json { ignoreUnknownKeys = true }
+        try {
+            val responseText = json.decodeFromString<Response>(recognizedText)
+            return Result(responseText.text, elapsedTimeInMs)
+        } catch (e: Exception) {
+            println("Error: ${e.message}")
+        }
+
+
+        val errorMsg = json.decodeFromString<ErrorResponse>(recognizedText)
+        return Result(errorMsg.error.message, elapsedTimeInMs)
```

Comment:
it seems odd to just return the error message as the result. might be better to signal that there was an error.

Reply:
Really the custom op needs to detect an error so the inference fails. Otherwise you can't add downstream nodes in the model as they'll be receiving bogus input. 

Given this is an example app, I'm fine with outputting the json as-is for now but if it's simple to add an alert or something feel free to.

Reply:
> Really the custom op needs to detect an error so the inference fails. Otherwise you can't add downstream nodes in the model as they'll be receiving bogus input.

Got it.

Agree that it doesn't make much sense to return an error message but with the app showing `successful recognition`. Will add an alert and mark it as Error in the app.

Reply:
Updated. Current behavior is inference result is allowed to be successful with an error response. The app will check if it's an error response. If so, show as `Error` and display the message from return result of openai endpoint:
![Screenshot 2023-09-15 at 11 10 46 AM](https://github.com/microsoft/onnxruntime-inference-examples/assets/35738743/1f168c19-d6d3-4ad4-ac8e-882fc37f6813)